### DISCUSSION TITLE
Installing Singular (computer algebra system) through MSYS2

### DISCUSSION THREAD

**Discussion by Zalnd** [msys2/MSYS2-packages#2433] [Q&A]
Is it possible to follow these install instructions in MSYS2 instead of cygwin?

```
1) install cygwin ( https://cygwin.com/setup-x86_64.exe) with default settings, additional,

at "Select Packages" choose wget.

2) in the cygwin terminal:

cd /

wget ftp://jim.mathematik.uni-kl.de/repo/cygwin/64/singular-4.2.0-1.tar.xz
tar Jxf singular-4.2.0-1.tar.xz

Now Singular can be started within the cygwin terminal as Singular.
```

Source: https://www.singular.uni-kl.de/index.php/singular-download/install-windows.html

Sorry for the noob question, but my knowledge of Linux related applications is close to zero. My regular use of msys2 is limited to compile C code that requires mpfr and gmp libraries.

**Comment by 1480c1** [comment]
In theory, unless it explicitly relies on something that cygwin provides but msys2 doesn't (an example being x11) it might work, but it would be best to go ahead and try, or alternatively to make sure it's "compatible", you could try seeing if compiling from the source might give you a different result

https://github.com/Singular/Singular

**Reply by Zalnd** [reply]
I tried to follow the instructions but the command `tar Jxf singular-4.2.0-1.tar.xz` seems to get stuck every time. I will forget Singular for now and move on to another free computer algebra system because building something from source is away out of my league.

In fact, that is exactly why I installed msys2 in the first place. hehe. I have failed to build both gmp and mpfr myself and then found out about msys2 and the packages `mingw-w64-x86_64-mpfr` and `mingw-w64-x86_64-gmp`. Running `pacman -S <package>` was a lot easier than the whole `./configure`-`make` process in which error messages spawn like heads of an hydra.

Thank you for your time anyway.

**Reply by goyalyashpal** [reply]
> move on to another free computer algebra system

hey @Zalnd which one did u pick?

**Reply by Zalnd** [reply]
> hey @Zalnd which one did u pick?

Cocoa provides a ready-to-use windows build with its own cygwin1.dll: https://cocoa.dima.unige.it/cocoa/download/install5-win.shtml.

But for as long as I can remember, I also ended setting up a virtual machine to install Singular via cygwin.

**Reply by goyalyashpal** [reply]
i wonder if [GNU Octave] could have helped u? Being compatible with Matlab, I'd think it to be fairly capable.
All these 3 are on the wikipedia [List_of_CAS] with following descriptions:

> Specialized CAS for [commutative algebra](https://en.wikipedia.org/wiki/Commutative_algebra)
\- CoCoA
>
> A high-level programming language for scientific computing and numerical computation ...
\- GNU Octave
>
> Computer algebra system for polynomial computations, with special emphasis on [commutative](https://en.wikipedia.org/wiki/Commutative_algebra) and [non-commutative algebra](https://en.wikipedia.org/wiki/Non-commutative_algebra), [algebraic geometry](https://en.wikipedia.org/wiki/Algebraic_geometry), and [singularity theory](https://en.wikipedia.org/wiki/Singularity_theory).
\- SINGULAR

[GNU Octave]: https://octave.org
[List_of_CAS]: https://en.wikipedia.org/wiki/List_of_computer_algebra_systems

**Reply by Zalnd** [reply]
> i wonder if [GNU Octave](https://octave.org) could have helped u? Being compatible with Matlab, I'd think it to be fairly capable.

Well, in theory, MATLAB provides the Symbolic Toolbox and Octave mimics that toolbox with its [symbolic](https://gnu-octave.github.io/packages/symbolic/) package (implemented using sympy).

But they are not focused on commutative algebra nor algebraic geometry. For example, they don't implement some state-of-art algorithms related to multivariate polynomial projection and the elimination theory. At least that was the situation two/three years ago.

**Reply by goyalyashpal** [reply]
> they don't implement some state-of-art algorithms related to ...

yeah, that's what i was thinking that it may be lacking in that sense. thanks a lot for your input :smiley:
